src/ng/directive/input.js

```diff
@@ -1540,24 +1540,34 @@ function createDateInputType(type, regexp, parseDate, format) {
     });
 
     if (isDefined(attr.min) || attr.ngMin) {
-      var minVal;
+      var minVal = attr.min || $parse(attr.ngMin)(scope);
+      var parsedMinVal = parseObservedDateValue(minVal);
+
       ctrl.$validators.min = function(value) {
-        return !isValidDate(value) || isUndefined(minVal) || parseDate(value) >= minVal;
+        return !isValidDate(value) || isUndefined(parsedMinVal) || parseDate(value) >= parsedMinVal;
       };
       attr.$observe('min', function(val) {
-        minVal = parseObservedDateValue(val);
-        ctrl.$validate();
+        if (val !== minVal) {
```

Comment:
So, if `val` is compared against `minVal` and `ctrl.$validate()` is only run inside `$observe()`, what guarantees that it is run at least once (for the initial `minVal`)?

I guess in most cases, `ngModel` itself runs it, but is it always guaranteed to happen after the `min` validator has been set up?

(I am not saying it isn't, just wondering.)

Reply:
For our validators, it's guaranteed that they will run when ngModel is set up. That's because we have a watch on the model value, so everything that is added in a link fn is guaranteed to be set up before the model value is first set to control / validated in the watch action fn.